plaidml/bridge/openvino/ops/deformable_convolution.cpp

```diff
@@ -0,0 +1,285 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "plaidml_ops.hpp"
+#include "plaidml_util.hpp"
+
+#include "ngraph/opsets/opset.hpp"
+#include "ngraph/opsets/opset4.hpp"
+
+#include "plaidml/op/op.h"
+
+using namespace plaidml;          // NOLINT[build/namespaces]
+using namespace InferenceEngine;  // NOLINT[build/namespaces]
+using namespace edsl;
+
+namespace {
+
+// Compute pad_before and the size of output.
+// This function is based on (the functon compute_padding_and_output_size() in plaidml/op/lib/ops.cc).
```

Comment:
Hmm, perhaps it would be possible to template the compute_padding_and_output_size function allowing it to take and produce `size_t` (or even something other integer like `int64_t`) instead of `TensorDim`. If that could be made to work that would be better as it would reduce code duplication.

Reply:
ok, I will submit another pr which templates the compute_padding_and_output_size function.

Reply:
We found out that we are able to reuse the original function in `plaidml/op/lib/ops.cc` in this op. We added the declaration in `plaidml/op/lib/ops.h` [(please see here)](https://github.com/plaidml/plaidml/pull/1711/files#diff-622f562b4f246e0fa4c4dc8fceb9be1e8e9ecd7109aa962e53c807591a7a43e6R40). But we also have to add  `plaidml::op::lib` as a dep of openvino plugin [(please see this line)](https://github.com/plaidml/plaidml/pull/1711/files#diff-56805e100050a20c6d5efc0b97e1ec1182f43cd55cccb1152e1e5545b961ab6bR22).

Reply:
@flaub Here's the function being used out of `plaidml::op::lib` that you asked about below (specifically `compute_padding_and_output_size`). It's used as a utility for `conv` and `pool` in the op library, and we need the same utility here for deformable convolution.

We could potentially expose this in `plaidml/op/op.h`, although it is a utility for writing operations that works on TensorDims, rather than an operation proper that works on Tensors like the rest of `op.h`. (Nevertheless, TensorDims can be wrapped as Values so I think exposing it like this should at least be possible.) We could also potentially move it to some shared utility location, although I'm not quite sure where that would be (a new `plaidml/op/util.h` perhaps?) or whether there would be additional problems from moving it to such a location.

Reply:
Given this extra complexity, I think we can return to the original design with  a DeformableConv-specific compute_padding function. We can keep in mind for the future that a possible improvement will be to reuse the code by following one of the paths from my previous comment.